app/src/main/java/ru/prometeydev/movie/model/MoviesPagingSource.kt

```diff
@@ -0,0 +1,31 @@
+package ru.prometeydev.movie.model
+
+import androidx.paging.PagingSource
+import retrofit2.HttpException
+import ru.prometeydev.movie.model.local.Movie
+import java.io.IOException
+import java.lang.Exception
+
+class MoviesPagingSource(
+    private val repository: MoviesRepository
+) : PagingSource<Int, Movie>() {
+
+    override suspend fun load(params: LoadParams<Int>): LoadResult<Int, Movie> {
+        val page = params.key ?: STARTING_PAGE_INDEX
+        return try {
+            val movies = repository.loadMovies(page)
+            LoadResult.Page(
+                data = movies,
+                prevKey = if (page == STARTING_PAGE_INDEX) null else page - 1,
+                nextKey = if (movies.isEmpty()) null else page + 1
```

Comment:
This is a bit un-efficient - you movie load has number of total pages. So you can just compare that as marker of the end.

Reply:
Yes, I agree, I will.